opentok/opentok.py

```diff
@@ -1639,46 +1641,88 @@ def __init__(
+        Mutes all streams in an OpenTok session.
+
+        You can include an optional list of streams IDs to exclude from being muted. 
+
+        :param session_id The session ID
 
-        :param session_id gets the session id from another function called get_session()
+        :param excludedStreamIds A list of stream IDs for streams that should not be muted. 
+        This is an optional property. If you omit this property, all streams in the session will be muted.
 
-        :param stream_id gets the stream id from another function called get_stream(). Note
-        that this variable is set to an empty string in the function definition as a specific
-        stream may not be chosen.
-   
+        :param active Whether streams published after the call, in addition to the current streams
+        in the session, should be muted (True) or not (False).
+
+        :param options is an empty dictonary representing the payload of the request,
+        whose contents will get populated inside the function
         """
-        
+
+        url = self.endpoints.get_mute_all_url(session_id)
+
         try:
-            if not stream_id:
-                url = self.endpoints.get_mute_all_url(session_id)
-                data = {'excludedStream': stream_id}
+            if excludedStreamIds:
+                if active:
+                    options = {'active': active, 'excludedStreams': excludedStreamIds }
             else:
-                url = self.endpoints.get_stream_url(session_id, stream_id) + "/mute"
-                data = None
-            
-            
-            response = requests.post(url, headers=self.get_headers(), data=data)
+                active = False
+                options = {'active': active, 'excludedStreams': []}
+  
+            response = requests.post(url, headers=self.get_headers(), data=json.dumps(options))
 
             if response:
                 return response
             elif response.status_code == 400:
                 raise GetStreamError("Invalid request. This response may indicate that data in your request data is invalid JSON. Or it may indicate that you do not pass in a session ID or you passed in an invalid stream ID.")
             elif response.status_code == 403:
-                raise AuthError("Failed to create session, invalid credentials")
+                raise AuthError("Failed to mute, invalid credentials.")
+            elif response.status_code == 404:
+                raise NotFoundError("The session or a stream is not found.")
+        except Exception as e:
+            raise OpenTokException(
+                ("There was an error thrown by the OpenTok SDK, please check that your session_id {0} and excludedStreamIds (if exists) {1} are valid").format(
+                    session_id, excludedStreamIds))
+
+        
+
+    def mute_stream(self, session_id: str, stream_id: str, options: dict = {}) -> requests.Response:
```

Comment:
See my comment about the `options` parameter of the `mute_all()` method.

Reply:
Ok, will do!